datalake/src/main/java/com/sequenceiq/datalake/service/validation/cloudstorage/CloudStorageValidator.java

```diff
@@ -106,14 +107,16 @@ public void validate(CloudStorageRequest cloudStorageRequest, DetailedEnvironmen
      * @param environment Details of the environment on which validation is performed.
      * @param validationResultBuilder Builder to gather the failures.
      */
-    public void validateBackupLocation(CloudStorageRequest cloudStorageRequest, DetailedEnvironmentResponse environment, String customBackupLocation,
+    public void validateBackupLocation(CloudStorageRequest cloudStorageRequest, BackupOperationType backupOperationType,
+            DetailedEnvironmentResponse environment, String customBackupLocation,
             ValidationResult.ValidationResultBuilder validationResultBuilder) {
         String backupLocation = !Strings.isNullOrEmpty(customBackupLocation) ? customBackupLocation : getBackupLocationBase(environment);
         LOGGER.info("Validating backup Location: {}", backupLocation);
         Credential credential = getCredential(environment);
         CloudCredential cloudCredential = credentialToCloudCredentialConverter.convert(credential);
         List<StorageLocationBase> locations = cloudStorageRequest.getLocations();
-        ObjectStorageValidateRequest request = createBackupLocationValidateRequest(cloudCredential, cloudStorageRequest, environment, backupLocation);
+        ObjectStorageValidateRequest request = createBackupLocationValidateRequest(cloudCredential, backupOperationType.name(), cloudStorageRequest,
```

Comment:
so I do not see any reason here to use String here for backupOperationType, instead of the enum, in the builder you are converting it back to enum

Reply:
BackupOperationType is defined in both datalake-api and cloudbreak-api. I have transalted it string to avoid conflict.

I addressed it by avoiding the need to define it twice.

Reply:
addressed